Add a blog post talking about the development focus areas for the next year (#172)
apple/swift-org-website: _posts/2022-11-17-focus-areas-next-year.md

* For [ownership](https://forums.swift.org/t/manifesto-ownership/5212), two things:

  - Developing features to give programmers explicit control over the ownership of values in memory. This includes features to prohibit implicit copies, transfer ownership between contexts, and explicitly “borrow” values without copying them.

  - Adding basic support for non-copyable types. This will provide new ways to achieve high performance by restricting the lifecycle of critical values. These controls will in turn enable new ways to work with data in memory, combining the performance of current “unsafe” constructs with the safety of Swift’s standard library features.

* To further empower the creation of rich libraries and DSLs, developing the basics of a [procedural macro system](https://forums.swift.org/t/a-possible-vision-for-macros-in-swift/60900).  This effort will start with the creation of a vision document that will lay out a design for what macros can achieve in Swift and how they might fit into the language.

> * Introducing support for [C++ interoperability](https://forums.swift.org/t/report-swift-and-c-interoperability-project-progress-in-the-swift-5-7-time-frame/61005) under the auspices of the new dedicate C++ Interoperability Workgroup:

Reviewer: Should `dedicate` here be `dedicated` or am I wrong? Isn't dedicate is a verb?

Reviewer 2: ```suggestion
* Introducing support for [C++ interoperability](https://forums.swift.org/t/report-swift-and-c-interoperability-project-progress-in-the-swift-5-7-time-frame/61005) under the auspices of the new dedicated C++ Interoperability Workgroup:
```

- Enhancing the quality of automatically generated Swift textual interfaces and the binary module infrastructure to help library authors to ship their Swift APIs more reliably and efficiently. Related diagnostics will also be added to nurture good engineering practices on that regard.

- Developing features to enable flexible software integration, allowing software components developed by different teams at different cadence to be integrated smoothly, both at compile time and at run time.

## Package registry

Development on the Swift package manager is focused on starting work on an open source package registry server implementation in concert with the community. The goal is to create the technical components required to transition Swift packages ecosystem from source control based dependencies to registry based ones, enhancing the security and reliability of the ecosystem.

## Implementation improvements

In addition to the feature work above, compiler developers are focused on several improvements at the implementation level:

> - Developing a Swift parser that's written in pure Swift and feature-complete with the current C++ implementation.  This opens up a lot of new directions in tooling development, and someday it may even be able to supplant the C++ parser.

Reviewer 2: ```suggestion
- Developing a Swift parser that's written in pure Swift and feature-complete with the current C++ implementation.  This opens up a lot of new directions in tooling development and, once at feature parity, will supplant the C++ parser.
```

"Some day may even be able" seems way over hedged. It _will_ replace the C++ parser.

## Core Team addition

The Swift Core Team will continue to focus on building out the community and supporting the growth of the Swift ecosystem.  [Mishal Shah](https://github.com/shahmishal) will join the Core Team to bring additional leadership to that critical direction.  As a lead and advocate for much of the underlying infrastructure that supports the Swift project, Mishal has played an essential role in supporting the Swift project's growth for some time.

## Summary

That's it for now!  We've tried to cover the most important work happening across the project, but there are a lot of smaller things going on, too, that we just didn't have space to cover.  We hope that putting this information together in one place like this has been useful.

> If you're interested in getting involved, there are a lot of ways to do it.  Language changes will be going through the normal evolution process, so keep an eye out for [discussion threads, pitches, and proposal reviews](https://forums.swift.org/c/evolution/18) about the work that you're interested in.  If you have questions or feedback about any of the items in this post, you can reach out to the appropriate workgroup, or, if you're not sure what that is, on the forums thread associated with this post (<link here>).

Reviewer 3: >...on the forums thread associated with this post (<link here>).

Just leaving this comment here so we don't forget to update this link placeholder. 🙂
